### PR TITLE
Add sample for color switch in colors docs

### PR DESCRIPTION
**Pull request by Karneades**
One of the things I used for a current project was a switch to change the color schemes quickly. I added the snippet in [Specifying Colors & Gradients](https://squib.readthedocs.io/en/latest/colors.html) and in the wiki: https://github.com/andymeneely/squib/wiki/Switch-card-background-or-invert-theme.

### CODE REVIEW

File: docs/colors.rst [hunk 1]
```diff
@@ -88,3 +88,42 @@ Sample: gradients
...
+
+Say you have different card types with different colors or themes but you would like to change the theme quickly without changing all parameters for each method inside ``Squib::Deck.new()``.
+
+You could use something like the following snippet to have one place to define the color/theme and use that in all map functions. The example inverts the color from white to black for one card type. Use a switch-statement inside the map function to differentiate multiple types.
+
+It's possible to use that for e.g. background color, text color or to choose the correct SVG for that color scheme (e.g. use the white or the black version of an image). The sample shows how to define the color at one place, e.g. choose between white and black, to quickly change the color scheme for the cards::
+
+  color = 'black'
+
+  background_color = all_cards['Type'].map { |t|
```
Comment (andymeneely) [inline docs/colors.rst]:
```suggestion
  background_color = all_cards['Type'].map do |t|
```
When the block is more than one line, the convention is to use `do`/`end`

File: docs/colors.rst [hunk 2]
```diff
@@ -88,3 +88,42 @@ Sample: gradients
...
+      if color == 'black' && t == "Type1" then
+          "black"
+      else
+          "white"
+      end
+  }
```
Comment (andymeneely) [inline docs/colors.rst]:
```suggestion
  end
```

File: docs/colors.rst [hunk 3]
```diff
@@ -88,3 +88,42 @@ Sample: gradients
...
+  background color: background_color
+
+  text_color = all_cards['Type'].map { |t|
```
Comment (andymeneely) [inline docs/colors.rst]:
```suggestion
  text_color = all_cards['Type'].map do |t|
```

File: docs/colors.rst [hunk 4]
```diff
@@ -88,3 +88,42 @@ Sample: gradients
...
+      if color == 'black' && t == "Type1" then
+          "white"
+      else
+          "black"
+      end
+  }
```
Comment (andymeneely) [inline docs/colors.rst]:
```suggestion
  end
```

File: docs/colors.rst [hunk 5]
```diff
@@ -88,3 +88,42 @@ Sample: gradients
...
+
+  text str: all_cards['Type'], layout: 'title', color: text_color
+
+  svg file: all_cards['Type'].map { |t|
```
Comment (andymeneely) [inline docs/colors.rst]:
```suggestion
  svg layout: 'illustration', file: all_cards['Type'].map do |t|
```

Comment (andymeneely) [reply docs/colors.rst]: I personally often miss it when there's a comma after an "end".

File: docs/colors.rst [hunk 6]
```diff
@@ -88,3 +88,42 @@ Sample: gradients
...
+      if color == 'black' && t == "Type1" then
+          "icons/#{t.downcase}-white.svg"
+      else
+          "icons/#{t.downcase}.svg"
+      end
+  }, layout: 'illustration'
```
Comment (andymeneely) [inline docs/colors.rst]:
```suggestion
  end
```

File: docs/colors.rst [hunk 7]
```diff
@@ -88,3 +88,42 @@ Sample: gradients
 .. raw:: html
 
   <img src="colors/gradient_00_expected.png" width=600 class="figure">
+
+Sample: Switch color based on variable
```
Comment (andymeneely) [inline docs/colors.rst]: Fantastic! Can you do the following:

- [ ]  put a working sample in `samples/colors` directory. And then you can include it here with an include in the same way we do in https://github.com/andymeneely/squib/blob/dev/docs/build_groups.rst - with "literalinclude". 
- [ ] name the sample file starting with an underscore

I run the samples directory on my regression testing, so if this ever breaks we'll know. 

I also made some suggestions to use more conventional formatting.

Comment (Karneades) [reply docs/colors.rst]: Thanks for the hint regarding adding a sample. I added the required files and I think you'll squash the commits on merge, right?

BTW I don't know how to include csv files into the docs like it is used in https://github.com/andymeneely/squib/blob/dev/docs/guides/getting-started/part_1_zero_to_game.rst#data-or-layout. I add a table to the docs with the used data from the sample.

**Pull request merged**